reinforcement_learning/rlclientlib/logger/event_batcher.cc

```diff
@@ -0,0 +1,53 @@
+#include "event_batcher.h"
+
+#include "generated/RankingEvent_generated.h"
+#include "generated/OutcomeEvent_generated.h"
+
+namespace reinforcement_learning {
+  namespace utility {
+
+    event_batcher::event_batcher()
+    {
+    }
+
+    void event_batcher::batch_serialize(data_buffer& oss, size_t& remaining, event_queue<ranking_event>& queue, size_t _send_high_water_mark)
+    {
+      batch_serialize_internal<ranking_event, RankingEvent, VW::Events::RankingEventBatchBuilder>(oss, remaining, queue, _send_high_water_mark, _batch_size);
+    }
+    void event_batcher::batch_serialize(data_buffer& oss, size_t& remaining, event_queue<outcome_event>& queue, size_t _send_high_water_mark)
+    {
+      batch_serialize_internal<outcome_event, OutcomeEvent, VW::Events::OutcomeEventBatchBuilder>(oss, remaining, queue, _send_high_water_mark, _batch_size);
+    }
+
+    template<typename TEvent, typename TResultObject, typename TResultObjectBuilder>
+    void event_batcher::batch_serialize_internal(data_buffer& oss, size_t& remaining, event_queue<TEvent>& queue, size_t _send_high_water_mark, size_t batch_size)
+    {
+      if (_batch_size <= 0) {
+        throw("event_batcher::batch_serialize batch_size is 0 - message is too large.");
+      }
+
+      std::vector<flatbuffers::Offset<TResultObject>> events_offset;
+      flatbuffers::FlatBufferBuilder builder(_send_high_water_mark);
+      TEvent evt;
+      for (int i = 0; remaining > 0 && oss.size() < _send_high_water_mark; i++) {
+        if (i == batch_size) break;
```

Comment:
Prefer to avoid break in the loop if you can. And if it is needed, split it into several lines and add a comment

Reply:
This is required. Please see detains in the discussion with Alexey.

Reply:
This is quite self-explained. It will loop to the batch_size defined in the code.